### ISSUE TITLE
Turnip Freedreno Android

### ISSUE THREAD

**Issue by klark231** [MastaG/mesa-turnip-ppa#2]
can you please release the vulkan.so for Android for use on Termux.

**Comment by MastaG** [comment]
You can download the patched turnip driver from the releases page.
Extract it and rename ```vulkan.ad07XX.so``` to ```libvulkan.so``` for use with Termux.
If you're running Ubuntu in proot/chroot then you can just use the PPA as described in the README

**Comment by klark231** [comment]
hmmmmmmmmmm but in most ubuntu chroot we use libvulkan_freedreno.so

**Comment by MastaG** [comment]
Yes, then just follow the instructions and add the launchpad ppa repository :)
You'll get the nightly dri3 patched turnip driver with kgsl enabled.

**Comment by klark231** [comment]
I tried to rename libvulkan.so to libvulkan_freedreno.so but it doesnt work game wont boot

**Comment by klark231** [comment]
nevermind tried the latest one it worked

**Comment by klark231** [comment]
oh well thought it worked but I was using another driver didnt notice. tried your again and wont do anything
![Uploading Screenshot_2024-07-25-18-22-31-166_com.termux.jpg…]()

**Comment by MastaG** [comment]
But are you using termux native or proot?

**Comment by klark231** [comment]
> But are you using termux native or proot?

chroot with ubuntu noble

**Comment by klark231** [comment]
> But are you using termux native or proot?

actually have you tried it with wine or only linux apps?

**Comment by klark231** [comment]
I tried to install proot-distro but the main problem is the icd.d its not there exporting it do nothing

**Comment by MastaG** [comment]
I've tried SuperTuxKart with both native vulkan and zink and vkmark and it all works fine on my S24.
This is Ubuntu 24.04 with proot.
It also works on my OnePlus 7T Pro in chroot.

But I'm using the driver from my PPA as per instructions of the readme.

For the record the PPA is nightly built from Oibaf's mesa (main git) with the dri3 patches applied and kgsl enabled.
No need to load custom libvulkan drivers as the PPA will upgrade the OS driver packages.

**Comment by klark231** [comment]
> I've tried SuperTuxKart with both native vulkan and zink and vkmark and it all works fine on my S24. This is Ubuntu 24.04 with proot. It also works on my OnePlus 7T Pro in chroot.
> 
> But I'm using the driver from my PPA as per instructions of the readme.
> 
> For the record the PPA is nightly built from Oibaf's mesa (main git) with the dri3 patches applied and kgsl enabled. No need to load custom libvulkan drivers as the PPA will upgrade the OS driver packages.

so I was able to make it work now. but it works a lot slower than other turnip drivers. others tested it too

you can try turnip by Mobox it works almost 2x faster than this for some reason

**Comment by MastaG** [comment]
Perhaps this is because I don't apply any other patches or open merge requests.
The PPA repository contains mesa from main git, with only the dri3 patch and the kgsl enabled.

**Comment by MastaG** [comment]
I just rebased the dri3 patch and launchpad built new packages for both 24.04 and 24.10.
Try doing apt update + upgrade.

**Comment by astroskyoffical** [comment]
Can this driver work at all for termux native?

**Comment by MastaG** [comment]
yes, I've tried it on both termux proot and chroot.
however keep in mind that these are nightly builds.. so stuff can break from time to time.

**Comment by astroskyoffical** [comment]
> yes, I've tried it on both termux proot and chroot. however keep in mind that these are nightly builds.. so stuff can break from time to time.

I see.  I'm basically trying to get hardware acceleration working in termux native with out proot. Iv done so much research and my research has landed here for something more updated and patched.  I'm still a noob and I'm trying hard to learn do you mind giving me a basic run down of how I can get this build of yours to work in the native termux environment at all?  There's freedrino but unfortunately it's not very updated to support 750 like this is.

**Comment by MastaG** [comment]
I have a Samsung S24 which features a A750 and that's already supported for a while now within mesa.
Oh wait, you said native?

I didn't try that, the PPA is build for Ubuntu 20.04 and 20.10, so they probably won't work with termux native due to linkage / libc differences, so it won't work.

Checkout the termux repo's, there should be one or two mesa packages, you could try to edit them so they build with the dri3.patch from my repo.

**Comment by astroskyoffical** [comment]
Yeah that sound like a good way to fix this issue. Basically just patch the ones that come from termux and make it work somehow.

**Comment by astroskyoffical** [comment]
Hey idk if this can help anyone or you @MastaG  but basically I got Qualcomm drivers working in termux native lol......and to see how huge the speed increase is an understatement.  This .Deb package is about to be pushed to termux and I thought I'd shared this discovery with you so you can work out it's kinks.   There's a few issues. Bender marks run amazing. I get almost 1k fps with glmark2 and vulkn tests.   However there's graphics issues and a few opengl translation issues 

I made a reddit post about it just now

https://www.reddit.com/r/termux/s/3NXgZwz1B8


Hopefully this can fixed!
